launcher/main.go

```diff
@@ -226,6 +245,18 @@ func readConfig() (conf Config) {
 		conf.FlatpakMetadataTemplate = flatpakMetadataTemplate
 	}
 
+	waylandProxyArgsJson, useWaylandProxy := os.LookupEnv("WAYLAND_PROXY_ARGS")
+	conf.UseWaylandProxy = useWaylandProxy
+	if useWaylandProxy {
+		conf.WaylandProxyArgs = readJsonArgs(waylandProxyArgsJson)
+		conf.WaylandProxyExe = envOr("WAYLAND_PROXY_EXE", "wayland-proxy-virtwl")
+		conf.WaylandProxySocketPath = getOptionValue(conf.WaylandProxyArgs, "--wayland-display")
```

Comment:
Probably also panic if `WaylandProxySocketPath` isn't set.

Reply:
Should be fixed by moving the whole option generation from launch.nix to launcher.